packages/site/cms-site/src/blocks/ExternalLinkBlock.tsx

```diff
@@ -1,23 +1,25 @@
 import * as React from "react";
 
 import { ExternalLinkBlockData } from "../blocks.generated";
-import { IFrameMessageType } from "../iframebridge/IFrameMessage";
-import { useIFrameBridge } from "../iframebridge/useIFrameBridge";
+import { usePreview } from "../preview/usePreview";
+import { SitePreviewIFrameMessageType } from "../sitePreview/iframebridge/SitePreviewIFrameMessage";
+import { useSitePreviewIFrameBridge } from "../sitePreview/iframebridge/useSitePreviewIFrameBridge";
 import { PropsWithData } from "./PropsWithData";
 
 interface ExternalLinkBlockProps extends PropsWithData<ExternalLinkBlockData> {
     children: React.ReactElement;
 }
 
 export function ExternalLinkBlock({ data: { targetUrl, openInNewWindow }, children }: ExternalLinkBlockProps): React.ReactElement {
-    const iframe = useIFrameBridge();
+    const iframe = useSitePreviewIFrameBridge();
+    const preview = usePreview();
 
-    if (iframe.hasBridge) {
+    if (preview.previewType === "SitePreview" || preview.previewType === "BlockPreview") {
```

Comment:
Shouldn't the onClick handler only be registered for the "SitePreview" preview type?

Reply:
fixed.

the onClick we still need - but only for preventDefault